pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -323,29 +334,68 @@ func makeUID() string {
 	return fmt.Sprintf("%08x", rand.Uint32())
 }
 
-// getTerminationMessage gets termination message of the container.
-func getTerminationMessage(status *runtimeapi.ContainerStatus, kubeStatus *kubecontainer.ContainerStatus, terminationMessagePath string) string {
-	message := ""
-
-	if !kubeStatus.FinishedAt.IsZero() || kubeStatus.ExitCode != 0 {
-		if terminationMessagePath == "" {
-			return ""
-		}
+const maxMessageLength = 1024 * 4
 
+// getTerminationMessage looks on the filesystem for the provided termination message path, returning a limited
+// amount of those bytes, or returns true if the logs should be checked.
+func getTerminationMessage(status *runtimeapi.ContainerStatus, terminationMessagePath string, fallbackToLogs bool) (string, bool) {
+	if len(terminationMessagePath) != 0 {
 		for _, mount := range status.Mounts {
-			if mount.GetContainerPath() == terminationMessagePath {
-				path := mount.GetHostPath()
-				if data, err := ioutil.ReadFile(path); err != nil {
-					message = fmt.Sprintf("Error on reading termination-log %s: %v", path, err)
-				} else {
-					message = string(data)
-				}
-				break
+			if mount.ContainerPath == nil || mount.HostPath == nil || *mount.ContainerPath != terminationMessagePath {
+				continue
+			}
+			path := *mount.HostPath
+			data, _, err := readLastBytesFromFile(path, maxMessageLength)
+			if err != nil {
+				return fmt.Sprintf("Error on reading termination log %s: %v", path, err), false
+			}
+			if !fallbackToLogs || len(data) != 0 {
+				return string(data), false
 			}
 		}
 	}
+	return "", fallbackToLogs
+}
+
+// readLastBytesFromFile reads at most max bytes from the provided file or returns an error.
+// It returns true if it the file was longer than max.
+func readLastBytesFromFile(path string, max int64) ([]byte, bool, error) {
```

Comment:
this is same with dockertools.readLastBytesFromFile, maybe move to `pkg/kubelet/util`?